theano/tensor/tests/test_basic.py

```diff
@@ -5188,6 +5189,37 @@ def test_infer_shape(self):
         assert numpy.all(f(0) == len(numpy.arange(0, 0)))
 
 
+class TestNdGrid(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def test_mgrid_numpy_equiv_float(self):
+        nfmgrid = numpy.mgrid[0:1:.1, 1:10:1., 10:100:10.]
+        tfmgrid = mgrid[0:1:.1, 1:10:1., 10:100:10.]
+        for ng, tg in zip(nfmgrid, tfmgrid):
+            assert_array_equal(ng, tg.eval())
+
+    def test_mgrid_numpy_equiv_int(self):
+        nimgrid = numpy.mgrid[0:2:1, 1:10:1, 10:100:10]
+        timgrid = mgrid[0:2:1, 1:10:1, 10:100:10]
+        for ng, tg in zip(nimgrid, timgrid):
+            assert_array_equal(ng, tg.eval())
+
+    def test_ogrid_numpy_equiv_float(self):
+        nfogrid = numpy.ogrid[0:1:.1, 1:10:1., 10:100:10.]
+        tfogrid = ogrid[0:1:.1, 1:10:1., 10:100:10.]
+        for ng, tg in zip(nfogrid, tfogrid):
+            assert_array_equal(ng, tg.eval())
+
+    def test_ogrid_numpy_equiv_int(self):
```

Comment:
Also, I do not see any tests with indices being Theano variable. Can you make one? No need to test all cases, but one where, start, stop and step are theano variable would be enough I think.